### PR TITLE
deps: update cel-cpp to 0.6.1

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit Message: update CEL evaluator to latest release
Risk Level: low
Testing: unit / regression
Docs Changes: none
Release Notes: none

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -89,12 +89,12 @@ REPOSITORY_LOCATIONS_SPEC = dict(
...
         project_url = "https://abseil.io/",
-        version = "5d8fc9192245f0ea67094af57399d7931d6bd53f",
-        sha256 = "e3812f256dd7347a33bf9d93a950cf356c61c0596842ff07d8154cd415145d83",
+        version = "a9831f1cbf93fb18dd951453635f488037454ce9",
+        sha256 = "8a144e21e0110903a1664dd89cabc327ce26bc56e04e02d49335544b9a64631f",
         strip_prefix = "abseil-cpp-{version}",
         urls = ["https://github.com/abseil/abseil-cpp/archive/{version}.tar.gz"],
         use_category = ["dataplane_core", "controlplane"],
-        release_date = "2020-11-24",
+        release_date = "2021-05-03",
```
Comment (moderation) [inline bazel/repository_locations.bzl:97]: Now that Abseil has regular releases, can / should we switch to the releases? https://github.com/abseil/abseil-cpp/releases

File: bazel/repository_locations.bzl [hunk 2]
```diff
@@ -705,8 +705,8 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Common Expression Language (CEL) C++ library",
         project_desc = "Common Expression Language (CEL) C++ library",
         project_url = "https://opensource.google/projects/cel",
-        version = "9841e3ee251f3cc4cd5b6dd9deee6818bc9f2854",
-        sha256 = "7e42cbad7d1068d6e7891ad101e2863e727692136d6b3a817c487b3cc7bcfdcc",
+        version = "3b7f7061d80cff882dc477faf74a2cd525ee9b62",
+        sha256 = "c4e38e791a6a56042cd15e06e025ab70e8ae0dcd2d7c3c3278c606f79b33f0eb",
```
Comment (moderation) [inline bazel/repository_locations.bzl:709]: As per the PR title can we switch to releases?
```
https://github.com/google/cel-cpp/archive/refs/tags/v0.5.0.tar.gz
293b4a0479a4390e08b24e7f14a307e4dd94495367b104c4a37926a760597aa7
2021-03-31
```

**Review by moderation — CHANGES_REQUESTED** [review]
Would be could to switch to releases for cel-cpp and abseil if possible

**Review by moderation — CHANGES_REQUESTED** [review]
As per comments it would be preferable to switch to releases for both cel-cpp and abseil

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (kyessenov) [reply bazel/repository_locations.bzl:97]: I'm not sure about this. Google just has a single master version in our monorepo, so not sure how the versions are decided.

File: bazel/repository_locations.bzl (hunk 2 above)
Comment (kyessenov) [reply bazel/repository_locations.bzl]: Done.

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (moderation) [reply bazel/repository_locations.bzl:97]: The repo recommends 'live-at-head' but document their release strategy at https://github.com/abseil/abseil-cpp#releases and https://abseil.io/about/releases

> NOTE: we don’t want you to think of these snapshots as “versions.” They are simply a snapshot of the codebase at a specific point in time.

Shall we stick with commit versions and 'live-at-head' for abseil @htuch ?

Comment (htuch) [reply bazel/repository_locations.bzl:113]: I think we prefer releases as an abstract handle, but if there are good functional or other reasons to not in a project with this philosophy, we can override.

**Review by htuch — DISMISSED** [review]
/lgtm deps

**Comment by kyessenov** [comment]
Windows compiler does not like initializing int64_t constexpr with long values, sending a patch upstream.

**Comment by kyessenov** [comment]
This might require some serious patching of cel-cpp since it assumes aliased string_views when using protobuf. Sigh.

**Comment by htuch** [comment]
@kyessenov is there any way to avoid Envoy-side patches? We're really trying to avoid these going forward.

**Comment by kyessenov** [comment]
@htuch Yes, it can be fixed upstream. It's difficult to enforce in g3, so it just needs regular maintenance to ensure abseil-protobuf use is valid with non-aliased absl and std string_views.

**Comment by wrowe** [comment]
Thank you for identifying the ambiguous long vs int64 issue. Awaiting cleaning up format check

**Comment by kyessenov** [comment]
The coverage dropped by 0.1% because of dependency upgrade? That is unexpected.

**Review by wrowe — APPROVED** [review]
LGTM /deps ... Unconcerned about third party code coverage

**Comment by kyessenov** [comment]
Yay, took 2 months on-off but finally ready!

**Pull request merged**